FirebaseRemoteConfig/SwiftNew/ConfigRealtime.swift

```diff
@@ -0,0 +1,667 @@
+  @objc func triggerListenerForTesting(listener: @escaping (RemoteConfigUpdate?, Error?) -> Void) {
+    DispatchQueue.main.async {
+      listener(RemoteConfigUpdate(), nil)
+    }
+  }
+
+  // MARK: - HTTP Helpers
+
+  private func appName(fromFullyQualifiedNamespace fullyQualifiedNamespace: String) -> String {
+    return String(fullyQualifiedNamespace.split(separator: ":").last ?? "")
+  }
+
+  private func reportCompletion(onHandler completionHandler: (
+    (RemoteConfigFetchStatus, Error?) -> Void
+  )?,
+  withStatus status: RemoteConfigFetchStatus,
+  withError error: Error?) {
+    guard let completionHandler = completionHandler else { return }
+    realtimeLockQueue.async {
+      completionHandler(status, error)
+    }
+  }
+
+  private func refreshInstallationsToken(completionHandler: (
+    (RemoteConfigFetchStatus, Error?) -> Void
+  )?) {
+    guard let installations, !options.gcmSenderID.isEmpty else {
+      let errorDescription = "Failed to get GCMSenderID"
+      RCLog.error("I-RCN000074", errorDescription)
+      settings.isFetchInProgress = false
+      reportCompletion(
+        onHandler: completionHandler,
+        withStatus: .failure,
+        withError: NSError(
+          domain: RemoteConfigErrorDomain,
+          code: RemoteConfigError.internalError.rawValue,
+          userInfo: [NSLocalizedDescriptionKey: errorDescription]
+        )
+      )
+      return
+    }
+
+    RCLog.debug("I-RCN000039", "Starting requesting token.")
+    installations.authToken { [weak self] result, error in
+      guard let self = self else { return }
+      if let error = error {
+        let errorDescription = "Failed to get installations token. Error : \(error)."
+        RCLog.error("I-RCN000073", errorDescription)
+        self.isRequestInProgress = false
+        var userInfo = [String: Any]()
+        userInfo[NSLocalizedDescriptionKey] = errorDescription
+        userInfo[NSUnderlyingErrorKey] = error
+
+        self.reportCompletion(
+          onHandler: completionHandler,
+          withStatus: .failure,
+          withError: NSError(domain: RemoteConfigErrorDomain,
+                             code: RemoteConfigError.internalError.rawValue,
+                             userInfo: userInfo)
+        )
+        return
+      }
+      guard let tokenResult = result else {
+        let errorDescription = "Failed to get installations token"
+        RCLog.error("I-RCN000073", errorDescription)
+        self.isRequestInProgress = false
+        reportCompletion(onHandler: completionHandler,
+                         withStatus: .failure,
+                         withError: NSError(domain: RemoteConfigErrorDomain,
+                                            code: RemoteConfigError.internalError.rawValue,
+                                            userInfo: [
+                                              NSLocalizedDescriptionKey: errorDescription,
+                                            ]))
+        return
+      }
```

Comment:
This was handled in one clause in the objc impl. The unwrapped error clause added an extra user info key, but it looks safe.

Reply:
No action required either. Just noting anything I see that differs..